apollo-client/src/main/java/com/ctrip/framework/apollo/internals/AbstractConfig.java

```diff
@@ -454,33 +451,74 @@ protected void clearConfigCache() {
     }
   }
 
-  protected void fireConfigChange(final ConfigChangeEvent changeEvent) {
-    for (final ConfigChangeListener listener : m_listeners) {
+  /**
+   * @param changes map's key is config property's key
+   * @return how many listener be fired
+   */
+  protected int fireConfigChange(String namespace, Map<String, ConfigChange> changes) {
+    final Set<String> changedKeys = changes.keySet();
+    final List<ConfigChangeListener> listeners = this.findMatchedConfigChangeListeners(changedKeys);
+
+    // notify those listeners
+    for (ConfigChangeListener listener : listeners) {
+      Set<String> interestedChangedKeys = resolveInterestedChangedKeys(listener, changedKeys);
+      InterestedConfigChangeEvent interestedConfigChangeEvent = new InterestedConfigChangeEvent(
+          namespace, changes, interestedChangedKeys);
+      this.notifyAsync(listener, interestedConfigChangeEvent);
+    }
+
+    return listeners.size();
+  }
+
+  /**
+   * Fire the listeners by event.
+   *
+   * @return how many listener be fired
+   */
+  protected int fireConfigChange(final ConfigChangeEvent changeEvent) {
```

Comment:
Will it break the compability when change return type from `void` to `int`?

Reply:
I think it will, since this method is protected, users may have overrode it